<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>J3DTab</class>
 <widget class="QWidget" name="J3DTab">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1095</width>
    <height>702</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" stretch="0,1">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_6">
     <item>
      <widget class="QPushButton" name="import_j3d">
       <property name="text">
        <string>Import J3D File</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="export_j3d">
       <property name="text">
        <string>Export J3D File</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="load_anim">
       <property name="text">
        <string>Load Animation</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <widget class="QWidget" name="verticalLayoutWidget_2">
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <widget class="QLineEdit" name="filter">
         <property name="placeholderText">
          <string>Filter</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTreeView" name="j3d_chunks_tree">
         <property name="minimumSize">
          <size>
           <width>370</width>
           <height>0</height>
          </size>
         </property>
        </widget>
       </item>
       <item>
        <widget class="AnimControl" name="joint_anim_control" native="true">
         <property name="anim_type" stdset="0">
          <string>joint</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="AnimControl" name="reg_anim_control" native="true">
         <property name="anim_type" stdset="0">
          <string>reg</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="AnimControl" name="texidx_anim_control" native="true">
         <property name="anim_type" stdset="0">
          <string>texidx</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="AnimControl" name="texmtx_anim_control" native="true">
         <property name="anim_type" stdset="0">
          <string>texmtx</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="AnimControl" name="vis_anim_control" native="true">
         <property name="anim_type" stdset="0">
          <string>vis</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="j3d_sidebar" native="true">
      <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,1">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="j3d_sidebar_label">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Extra information will be displayed here as necessary.</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="toggle_visibility">
           <property name="maximumSize">
            <size>
             <width>60</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="update_j3d_preview">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>30</width>
             <height>16777215</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QScrollArea" name="scrollArea">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>5</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>250</width>
           <height>0</height>
          </size>
         </property>
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="scrollAreaWidgetContents">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>393</width>
            <height>618</height>
           </rect>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3"/>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="verticalLayoutWidget">
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QScrollArea" name="j3dultra_error_area">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="scrollAreaWidgetContents_2">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>298</width>
            <height>68</height>
           </rect>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="j3dultra_error_label">
             <property name="text">
              <string>No errors to display.</string>
             </property>
             <property name="wordWrap">
              <bool>true</bool>
             </property>
             <property name="textInteractionFlags">
              <set>Qt::LinksAccessibleByMouse|Qt::TextSelectableByMouse</set>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
       <item>
        <widget class="J3DViewer" name="j3d_viewer">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
           <horstretch>0</horstretch>
           <verstretch>5</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>300</width>
           <height>300</height>
          </size>
         </property>
         <property name="focusPolicy">
          <enum>Qt::ClickFocus</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
  <action name="actionOpenJ3DImage">
   <property name="text">
    <string>Open Image</string>
   </property>
  </action>
  <action name="actionReplaceJ3DImage">
   <property name="text">
    <string>Replace Image</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>J3DViewer</class>
   <extends>QOpenGLWidget</extends>
   <header>gcft_ui.j3d_viewer</header>
  </customwidget>
  <customwidget>
   <class>AnimControl</class>
   <extends>QWidget</extends>
   <header>gcft_ui.anim_control</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
